docs/design/editorial_design.md

```diff
@@ -0,0 +1,354 @@
+        items around it.
+    delta: The delta that we're looking to push this item about.
+"""
+otio.algorithms.slide(
+    item: Item,
+    delta: RationalTime, # Relative like slide
+)
+```
+- In the example image, `C` is `item` and `delta` is the arrow's vector. The edit command does the rest of the item management for us.
+
+---
+
+## Ripple
+![Ripple](../_static/edit/07_ripple.png)
+- Adjust a source_range without adjusting any other items
+- This effectively shifts all currently adjacent items to stay at the edges
+- No items _before_ the item are moved/affected
+- (Impl detail for otio, this is the same as adjusting the source_range of the item)
+
+#### API
+```py
+"""
+Args:
+    item: The item to initiate the edit on
+    delta_in: see trim()
+    delta_out: see trim()
+"""
+otio.algorithms.ripple(
+    item: Item,
+    delta_in: RationalTime = None,   #< Duration of change
+    delta_out: RationalTime = None,  #< Duration of change
+)
+```
+- In the example image, `A` is item and `delta_out` is a negative RationalTime. The edit command shifts `B` without adjusting it's source_range.
+
+> Again, no crazy about the argument names
+
+---
+
+
+## Roll
+![Roll](../_static/edit/09_roll.png)
+- Any trim-like action results in adjacent items source_range being adjusted to fit
+- No new items are ever created
+- Clamped to available media (if available)
+
+#### API
+````py
+"""
+Args:
+    item: The item to initiate the edit on
+    delta_in: see trim()
+    delta_out: see trim()
+"""
+otio.algorithms.roll(
+    item: Item,
+    delta_in: RationalTime = None,   #< Duration of change
+    delta_out: RationalTime = None,  #< Duration of change
+)
+````
+- In the example image, `A` is the item and `delta_out` is a negative RationalTime. The edit command augments `B` without changing it's `source_range().end_time_exclusive()`
+- If `delta_in` is supplied, the item to the left of `A` would be modified (if any) - otherwise we may need to add a `Gap` / fill template.
+
+---
+
+## 3/4 Point Edit
+![3_4 Point Edit](../_static/edit/08_pointedit.png)
+- The most complex - this "fills" a gap based on a source in/out point _or_ track in/out point
+- Often used to patch in items as edit is built
+- Note: This can be accomplished by a conjunction of commands above
+
+### API
+```py
+"""
+Args:
+    item: The item to place onto the track
+    track: Track that will now own this item
+    replace: Item (or RT?) that we will effectively replace
+    reference_point: For 4 point editing, the reference point dictates what
+        transform to use when running the fill.
+
+        Options:
+            - Source   : Don't modify the source, overwrite if required
+            - Sequence : Don't modify the sequence, trim item if required
+            - Fit      : Apply Time Effect
+"""
+otio.algorithms.fill(
+    item: Item,
+    track: Track,
+    replace: Item = None, # Alternatively, we could have a parent-coord
```

Comment:
This feels like this should maybe be a `TimeRange` within the track. If there is just a single item to operate on the caller can use `range_in_parent` to get the value they want to use.